mgr/manager.go

```diff
@@ -239,27 +247,39 @@ func (csm *CertServiceManager) CheckDiskPKI() error {
 // should not be constructed by hand.
 type Manager struct {
 	// Dir is the directory containing the certificate specs.
-	Dir string `json:"certspecs" yaml:"certspecs"`
+	Dir string `yaml:"certspecs"`
 
 	// DefaultRemote is used as the remote CA server when no
 	// remote is specified.
-	DefaultRemote string `json:"default_remote" yaml:"default_remote"`
+	DefaultRemote string `yaml:"default_remote"`
 
 	// ServiceManager is the service manager used to restart a
 	// service.
-	ServiceManager string `json:"service_manager" yaml:"service_manager"`
+	ServiceManager string `yaml:"service_manager"`
 
 	// Before is how long before the cert expires to start
 	// attempting to renew it.
-	Before string `json:"before" yaml:"before"`
-	before time.Duration
+	Before time.Duration `yaml:"before"`
 
 	// Interval is how often to update the NextExpires metric.
-	Interval string `json:"interval" yaml:"interval"`
-	interval time.Duration
+	Interval time.Duration `yaml:"interval"`
 
 	// Certs contains the list of certificates to manage.
-	Certs []*CertServiceManager `json:",omitempty" yaml:",omitempty"`
+	Certs []*CertServiceManager `yaml:",omitempty"`
+}
+
+func (m *Manager) UnmarshallYAML(unmarshall func(interface{}) error) error {
+	m = &Manager{
+		Before:   DefaultBefore,
+		Interval: DefaultInterval,
+	}
+	// use a cast to prevent unmarshall from going recursive against this
+	// deserializer function.
+	type plain Manager
+	if err := unmarshall((*plain)(m)); err != nil {
+		return err
+	}
+	return nil
 }
 
 // NewFromConfig loads a new Manager from a config file. This does not load the
```

Comment:
So from my understanding, NewFromConfig() is never actually used. Not sure if coming in a future PR, but worth investigating if it's also dead code/worth ripping out

Reply:
that's... weird.

Quick tracing of cli/root, I get the feeling this is the 'new' function that was added but never properly converted over to.

Frankly I'm not even sure where the hell certmgr.yaml gets read right now, but I'll run it down.

Reply:
I believe it's in root.go: `initConfig()`
`if err := viper.ReadInConfig(); err == nil {`

Reply:
Config is loaded via viper- which isn't desirable since viper just assumes the config is a key/value.  I'd prefer leaving this function in place, but I'll unfuck the config handling in a follow up PR.

Reply:
sgtm, lmk if you need any help with that/anything i can do

Reply:
If you've got a go library that integrates environment variables- cleanly- I'd be interested.  This viper setup is making me want to wipe it and replace it with just raw flags usage....

Reply:
[envy](https://github.com/jamiealquiza/envy) is pretty cool

Reply:
First glance, that gets my vote- it also has envy support, and is a helluva lot cleaner then this viper setup for extensiblity.

Reply:
I'll follow up on this in a seperate PR; the current docs imply to people that all configs are json, but in reality the certmgr.yaml is actually a key: value parse, and spec's can be json or yaml.